src/tensors/cpu/expression_graph_packable.h

```diff
@@ -200,25 +161,81 @@ class ExpressionGraphPackable : public ExpressionGraph {
+                                                     cols(val));
+          } else {
+            ABORT_IF(gemmElementType != Type::intgemm8, "Type {} is not supported {}", gemmElementType); // shouldn't really happen, but let's make sure
+            intgemm::Int8::PrepareA(tmp->data(), /*input*/
+                                    paramMat->data<int8_t>(), /*output*/
+                                    quantMult, /*Quant Mult*/
+                                    rows(val),
+                                    cols(val));
+          }
           //Put the quantMult at the back of the tensor
           *(reinterpret_cast<float *>(paramMat->data<int8_t>() + val->shape().elements())) = quantMult;
-        } else {
+
+        } else if(sizeOf(gemmElementType) == 2) { // is 16-bit Intgemm type
           float quantMult = 1024.0f;
-          intgemm::Int16::PrepareA(tmp->data(), /*input*/
-                                paramMat->data<int16_t>(), /*output*/
-                                quantMult, /*Quant Mult*/
-                                rows(val),
-                                cols(val));
+
+           // Hardware-specific conversions which allow to implement memory-mapping and avoid conversion at runtime
+          if(isSse2(gemmElementType)) {
+            // @TODO: we should croak here if the hardware type is not supported?
+            intgemm::SSE2_16bit::PrepareBTransposed(tmp->data(), /*input*/
+                                                    paramMat->data<int16_t>(), /*output*/
+                                                    quantMult, /*Quant Mult*/
+                                                    rows(val),
+                                                    cols(val));
+          } else if(isAvx2(gemmElementType)) {
+            // @TODO: we should croak here if the hardware type is not supported?
+            intgemm::AVX2_16bit::PrepareBTransposed(tmp->data(), /*input*/
+                                                    paramMat->data<int16_t>(), /*output*/
+                                                    quantMult, /*Quant Mult*/
+                                                    rows(val),
+                                                    cols(val));
+          } else if(isAvx512(gemmElementType)) {
+            // @TODO: we should croak here if the hardware type is not supported?
+            intgemm::AVX512_16bit::PrepareBTransposed(tmp->data(), /*input*/
+                                                      paramMat->data<int16_t>(), /*output*/
+                                                      quantMult, /*Quant Mult*/
+                                                      rows(val),
+                                                      cols(val));
+          } else {
+            ABORT_IF(gemmElementType != Type::intgemm8, "Type {} is not supported {}", gemmElementType); // shouldn't really happen, but let's make sure
```

Comment:
The surplus `{}` causes a formatting error before the abort.

Reply:
Also, it should be `gemmElementType != Type::intgemm16` here, otherwise I'm getting:

    Error: Type intgemm16 is not supported

when converting a model using `intgemm16`:

    marian-conv -f model.npz -t model.bin --gemm-type intgemm16